deep_quoridor/src/agents/greedy.py

```diff
@@ -97,18 +105,20 @@ def _shortest_path_from(self, game: Quoridor, pos: Position, target_row: int) ->
 
         # Trace back the path
         path = [dest]
-        while True:
+        prev_pos = None
+        while prev_pos != pos:
             last_pos = path[-1]
             prev_pos = coming_from[last_pos]
-            if prev_pos == pos:
-                break
-
```

Comment:
The change here does what @muralx  did of adding the initial position to the path